packages/headless/src/composables/helpers/useGlobalListener.ts

```diff
@@ -0,0 +1,71 @@
+import { onBeforeMount, onBeforeUnmount, reactive, watch, type Ref } from "vue";
+
+type DocumentEventType = keyof DocumentEventMap;
+type GlobalListener<K extends DocumentEventType = DocumentEventType> = (
+  event: DocumentEventMap[K],
+) => void;
+
+export type UseGlobalEventListenerOptions<K extends DocumentEventType> = {
+  type: K;
+  listener: GlobalListener<K>;
+  disabled?: Ref<boolean>;
+};
+
+const GLOBAL_LISTENERS = reactive(new Map<DocumentEventType, Set<GlobalListener>>());
```

Comment:
This is global state, right? I am not sure if this behaves correctly when used with SSR (e.g. Nuxt), so it might be shared across all clients

Reply:
Yes, it's global state. But listeners are only added on the client onBeforeMount, so everything should be fine in that regard.